### PR TITLE
Bugfix/#53-fix-branchprotection-bug

### LINKED ISSUE #53: Fixing branch protection fails when no protection found
```sh
% go run ./... configure-repo script-languages-release-c4 --fix

https://github.com/exasol/script-languages-release-c4
Warning: Failed to parse workflow definition 'https://github.com/exasol/script-languages-release-c4/blob/main/.github/workflows/build.yml'. Probably you use some advanced matrix build features there. Github-keeper will not add the checks from this workflow to the branch protection. Please add them manually. 
Warning: Failed to parse workflow definition 'https://github.com/exasol/script-languages-release-c4/blob/main/.github/workflows/release.yml'. Probably you use some advanced matrix build features there. Github-keeper will not add the checks from this workflow to the branch protection. Please add them manually. 
Warning: Failed to parse workflow definition 'https://github.com/exasol/script-languages-release-c4/blob/main/.github/workflows/test-release.yml'. Probably you use some advanced matrix build features there. Github-keeper will not add the checks from this workflow to the branch protection. Please add them manually. 
panic: Failed to create branch protection for exasol/script-languages-release-c4/main. Cause: PUT https://api.github.com/repos/exasol/script-languages-release-c4/branches/main/protection: 422 Invalid request.

No subschema in "anyOf" matched.
No subschema in "oneOf" matched.
Not all subschemas of "allOf" matched.
For 'anyOf/1', {"strict"=>false} is not a null. []

goroutine 1 [running]:
github.com/exasol/github-keeper/cmd.FixBranchProtectionProblemHandler.createBranchProtection({0x7ff7bfeff6a9?}, {0x7ff7bfeff6a9, 0x1b}, {0xc00020d7d0, 0x4}, 0x7ff7bfeff6a9?)
        /Users/chp/git/github-keeper/cmd/branchProtection.go:40 +0x205
github.com/exasol/github-keeper/cmd.BranchProtectionVerifier.CheckIfBranchProtectionIsApplied({{0x7ff7bfeff6a9?, 0xc0000c0008?}, 0xc000144000?}, 0x1e?)
        /Users/chp/git/github-keeper/cmd/branchProtection.go:57 +0x35b
github.com/exasol/github-keeper/cmd.glob..func1(0x18c91e0?, {0xc0000e24a0, 0x1, 0x2?})
        /Users/chp/git/github-keeper/cmd/configureRepo.go:29 +0x15f
github.com/spf13/cobra.(*Command).execute(0x18c91e0, {0xc0000e2460, 0x2, 0x2})
        /Users/chp/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x18c9960)
        /Users/chp/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/chp/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902
github.com/exasol/github-keeper/cmd.Execute()
        /Users/chp/git/github-keeper/cmd/root.go:16 +0x25
main.main()
        /Users/chp/git/github-keeper/main.go:6 +0x17
exit status 2
```

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #53

### CODE REVIEW

File: cmd/branchProtection.go [hunk 1]
```diff
@@ -80,9 +80,15 @@ func (verifier BranchProtectionVerifier) getRepo() *github.Repository {
...
 func (verifier BranchProtectionVerifier) addExistingChecksToRequest(existingProtection *github.Protection, protectionRequest github.ProtectionRequest) {
-	if existingProtection == nil || existingProtection.RequiredStatusChecks == nil || existingProtection.RequiredStatusChecks.Contexts == nil {
+	if existingProtection == nil || existingProtection.RequiredStatusChecks == nil || existingProtection.RequiredStatusChecks.Contexts == nil || len(existingProtection.RequiredStatusChecks.Contexts) == 0 {
 		return
 	}
+	if protectionRequest.RequiredStatusChecks == nil {
+		protectionRequest.RequiredStatusChecks = &github.RequiredStatusChecks{
+			Strict:   true,
+			Contexts: []string{},
+		}
```
Comment (jakobbraun) [inline cmd/branchProtection.go:90]: I'd expect that that's already the default. Maybe you could just wrap the for loop with:
```
if protectionRequest.RequiredStatusChecks != nil {
```

File: cmd/branchProtection.go [hunk 2]
```diff
@@ -185,6 +189,17 @@ func (verifier BranchProtectionVerifier) createProtectionRequest(requireSonar bo
...
 }
 
+func createRequiredStatusChecks(requiredChecks []string) *github.RequiredStatusChecks {
+	if len(requiredChecks) > 0 {
+		return &github.RequiredStatusChecks{
+			Strict:   true,
+			Contexts: requiredChecks,
+		}
+	} else {
+		return nil
```
Comment (jakobbraun) [inline cmd/branchProtection.go:199]: Why do we need nil here? Can' we return the RequiredStatusChecks with empty context list?

File: cmd/configureRepo.go [hunk 3]
```diff
@@ -23,14 +23,15 @@ var configureRepoCmd = &cobra.Command{
...
+		for index, repo := range args {
+			fmt.Printf("\nRepo %d of %d: https://github.com/%v/%v\n", index+1, len(args), org, repo)
 			branchProtectionVerifier := BranchProtectionVerifier{client: client, repoName: repo}
 			branchProtectionVerifier.CheckIfBranchProtectionIsApplied(fix)
 			UnifyLabels(repo, client, fix)
-			settingsVerifier := RepoSettingsVerifier{repo: repo, githubClient: client, org: "exasol"}
+			settingsVerifier := RepoSettingsVerifier{repo: repo, githubClient: client, org: org}
 			settingsVerifier.VerifyRepoSettings(fix)
-			webHookVerifier := WebHookVerifier{secrets: secrets, repo: repo, githubClient: client, org: "exasol"}
+			webHookVerifier := WebHookVerifier{secrets: secrets, repo: repo, githubClient: client, org: org}
```
Comment (jakobbraun) [inline cmd/configureRepo.go:34]: :+1:

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: cmd/branchProtection.go (hunk 1 above)
Comment (kaklakariada) [reply cmd/branchProtection.go:90]: Sorry, that's not possible with the GH API

File: cmd/branchProtection.go (hunk 2 above)
Comment (kaklakariada) [reply cmd/branchProtection.go:199]: Sorry, that's not possible with the GH API

**Review by jakobbraun — APPROVED** [review]

**Pull request merged**